### ISSUE TITLE
Is this project being maintained?

### ISSUE THREAD

**Issue by creeper-0910** [gunthercox/ChatterBot#2311]
Is this project being maintained?
Is it still available...?

**Comment by erickythierry** [comment]
**+1**

> _I also want to know_

**Comment by bbartling** [comment]
+2

**Comment by erickythierry** [comment]
apparently not 😕

**Comment by bbartling** [comment]
Are there any other similar projects like this that are not a cloud based
AI platform that requires a membership?

**Comment by beecode1001** [comment]
which dependencies have been added? It show errors on my side from the chatterbot after I have run and started the debugging process

**Comment by beecode1001** [comment]
I don't think GBT should be added to this project.I think Simple written lines of code should be made.

**Comment by jouky2** [comment]
the last commit was in 2021, so it's abandoned

**Comment by bbartling** [comment]
Dang, is everyone just using chat GPT?

What about scenarios where chat GPT may not be allowed due to some
companies policies...what are people using in that scenario?

**Comment by creeper-0910** [comment]
I was trying to learn a specific character using ChatterBot.
Is there any other software that can learn a specific character's personality, etc. and have them talk to each other in a local environment?

**Comment by jouky2** [comment]
Yes. You can use ora.ai and use a specific prompt. Use chatgpt but I mean you can get it to be rude or something so Who cares

**Comment by bbartling** [comment]
Cool...can you train ora.ai locally like you could with ChatterBot or is it
only a cloud application? I was on the chatterbot band wagon too late but I
thought it is cool that could potentially train on sensitive information
you would not want to share to some AI service like a chat GPT

**Comment by jouky2** [comment]
Well you could train gpt-2 yourself I believe you can have it locally but dont know if you can train it yourself. I know there's a openai model wich you can train yourself, but it's still on the server and used for chatgpt training so yeah. Also you can't even install this from pypi so maybe one of the countless forks does work.

**Comment by tommickey2020** [comment]
+1

**Comment by bbartling** [comment]
This stuff seems way cool...maybe there is a reason why this repo isnt being updated. The second video is more the basics of lang chain have you dabbled with this at all or aut GPT? https://youtu.be/rFQ5Kmkd4jc This second video is more the basics. https://youtu.be/aywZrzNaKjs

**Comment by lucasalberto01** [comment]
I still prefer this library over GPT chat for two reasons
1. This library is trained while you talk to it, gpt, if you train your own model, you first need the data and then leave it running for days to get the correct training.
2. With this library you can easily edit what was previously learned. in gpt you need to retrain the whole model.

For me they are two different tools, GPT is good for large texts, this one is better for chat.

I created a more modern fork and I'm trying to improve https://github.com/lucasalberto01/chatterbotpro

**Comment by erickythierry** [comment]
> 

another advantage of this lib: it's free, open source and self hosted.
Great for small and hobby projects.
I tried using gpt for some projects but it bothers me that I don't control the data that is fed to gpt and I still have to pay for each request

**Comment by bbartling** [comment]
Cool thanks 👍

**Comment by Uysim** [comment]
+1

**Comment by erickythierry** [comment]
> You can also use gpt-2 and train it yourself

The difference in complexity between the two is enormous. (not to mention the disparity in hardware resource usage to run gpt2)

**Comment by bbartling** [comment]
Does anyone know of a good repo to train open source LLM on your own data
set?

This is also a cool repo to run I think it's Facebook's Lama LLM in a
compressed model format on a CPU. It's a little slow but works. The model
is a 4 gigs .bin file that downloads and runs locally, it would be really
neat if there was something like this that could be customized with your
own data.

https://github.com/awinml/llama-cpp-python-bindings

**Comment by jdsayle** [comment]
For anyone interested in a maintained fork of Chatterbot, there are a few out there. The one that seems to be the most actively maintained (at least that I've come across) is here - https://github.com/ShoneGK/ChatterPy - last updates were 5months ago as of 8/25/2023 and it should work with Python 3.11 and up

**Comment by gunthercox** [comment]
Hey all (maintainer here) 👋🏼 

First off I really appreciate the kind words in a lot of these comments. ChatterBot being simple, free, and easy to use were always important aspects of the project to me and I'm glad to see that was valuable to so many people.

I had to take a break from working on this project, but as of late I've been working on getting it updated and fixed up where needed.

The AI space has changed significantly in the last few years and I'm hoping that the future of ChatterBot might be to continue to work simply but also to support more advanced features like large language models.

Anyway, I'm looking forward to working on things again. Keep an eye out for upcoming releases.

**Comment by creeper-0910** [comment]
> Hey all (maintainer here) 👋🏼
> 
> First off I really appreciate the kind words in a lot of these comments. ChatterBot being simple, free, and easy to use were always important aspects of the project to me and I'm glad to see that was valuable to so many people.
> 
> I had to take a break from working on this project, but as of late I've been working on getting it updated and fixed up where needed.
> 
> The AI space has changed significantly in the last few years and I'm hoping that the future of ChatterBot might be to continue to work simply but also to support more advanced features like large language models.
> 
> Anyway, I'm looking forward to working on things again. Keep an eye out for upcoming releases.

I am so glad you are back. 
Thank you !
